cmake/Gems.cmake

```diff
@@ -252,3 +252,77 @@ function(ly_enable_gems_delayed)
         endforeach()
     endforeach()
 endfunction()
+
+#! ly_is_gem_available: Check if a gem is available in the current project configuration.
+# \arg:NAME name of the gem to check for availability.
+# \arg:RETURN_VARIABLE name of the variable in which to set the returned value.
+# \arg:VALUE_IF_FOUND value to set in the RETURN_VARIABLE when the gem is available. Defaults to TRUE.
+# \arg:VALUE_IF_NOT_FOUND value to set in the RETURN_VARIABLE when the gem is not available. Defaults to FALSE.
+# \arg:VARIANT name of the project variant if which to check for availability. Defaults to 'Clients'.
+function(ly_is_gem_available)
+    set(options)
+    set(oneValueArgs NAME RETURN_VARIABLE VALUE_IF_FOUND VALUE_IF_NOT_FOUND VARIANT)
+    set(multiValueArgs)
```

Comment:
The VALUE_IF_FOUND and VALUE_IF_NOT_FOUND seems a little over-complex when it comes to returning whether the gem variant was found or not.
Following the cmake of having an `OUTPUT_VARIABLE` argument that either returns TRUE or FALSE would be simpler.
See the [cmake_path](https://cmake.org/cmake/help/latest/command/cmake_path.html?highlight=cmake_path#is-prefix) commands.

I don't foresee a situation where the `VALUE_IF_FOUND` and `VALUE_IF_NOT_FOUND` options need to ever be set.

Reply:
😅 Ok I will update that. My try was to allow the caller to customize the return value according to the availability, like for example:

```cmake
ly_is_gem_available(
    NAME EMotionFX
    RETURN_VARIABLE EMOTIONFX_DEF
    VALUE_IF_FOUND EMOTIONFX_ENABLED
    VALUE_IF_NOT_FOUND EMOTIONFX_DISABLED
    VARIANT Clients
)
```

And by default it acts like the cmake_path command:

```cmake
ly_is_gem_available(
    NAME EMotionFX
    RETURN_VARIABLE EMOTIONFX_AVAILABILITY # <- TRUE if available, FALSE otherwise
    VARIANT Clients
)
```